src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -18,23 +35,28 @@ export const TodoFilter: React.FC = () => {
 
       <p className="control is-expanded has-icons-left has-icons-right">
         <input
+          value={query}
           data-cy="searchInput"
           type="text"
           className="input"
           placeholder="Search..."
+          onChange={handleChangeQuery}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
-        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
-        </span>
+        {query && (
```

Comment:
```suggestion
        {!!query && (
```
in rendering with &&  need to convert the value to boolean.